### PR TITLE
fix: trip status changes in planner and trips

### PR DESCRIPTION
**Pull request by FH-sandeepkaur1**
## Change Description

trip status changes in planner and trips

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist: (Check only applicable)

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/app/pages/dispatch/dispatch.module.ts [hunk 1]
```diff
@@ -106,21 +106,36 @@ export class CustomAdapter extends NgbDateAdapter<string> {
...
   }
 
-  toModel(date: NgbDateStruct | null): string | null {
-    return date ? date.day + this.DELIMITER + date.month + this.DELIMITER + date.year : null;
+  toModel(date: NgbDateStruct): string // from internal model -> your mode
+  {
+    return date ? date.year + this.DELIMITER + ('0' + date.month).slice(-2)
+      + this.DELIMITER + ('0' + date.day).slice(-2) : null
   }
+  // toModel(date: NgbDateStruct | null): string | null {
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/dispatch.module.ts]: remove commented code if not required.

File: src/app/pages/dispatch/home/dispatch-overview/dispatch-overview.component.ts [hunk 2]
```diff
@@ -165,19 +165,19 @@ export class DispatchOverviewComponent implements OnInit {
       subscribe((result: any) => {
         for (let i = 0; i < result.Items.length; i++) {
           const element = result.Items[i];
-          if(element.isDeleted === 0 && element.tripStatus === 'planned'){
-            for (let j = 0; j < element.tripPlanning.length; j++) {
-              const element2 = element.tripPlanning[j];
-              if(element2.date != '' && element2.date != undefined) {
-                let pickDate = element2.date.split("-");
+          if(element.isDeleted === 0 && element.tripStatus === 'confirmed'){
+            // for (let j = 0; j < element.tripPlanning.length; j++) {
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/home/dispatch-overview/dispatch-overview.component.ts]: Remove commented code if not required.

File: src/app/pages/dispatch/planner/calendar-view/calendar-view.component.ts [hunk 3]
```diff
@@ -280,20 +283,22 @@ export class CalendarViewComponent implements OnInit {
...
       subscribe((result: any) => {
         result = result.Items[0];
         delete result.timeCreated;
         delete result.timeModified;
         this.tripData = result;
-        
+        this.OrderIDs = this.tripData['orderId'];
+
         if(this.tripData.tripPlanning.length === 0) { 
           this.toastr.error('Trip plan for selected trip is empty. Please create one to assign');
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/planner/calendar-view/calendar-view.component.ts]: Please update the sentence to:
Trip plan for selected trip is empty. Please create one to assign
TO 
The trip plan for the selected trip is empty. Please create one to assign.

File: src/app/pages/dispatch/trips/trip-detail/trip-detail.component.ts [hunk 4]
```diff
@@ -81,19 +81,19 @@ export class TripDetailComponent implements OnInit {
...
-              locationName: '',
-              stateID: element.location.stateID,
-              stateName: '',
-              zipcode: element.location.zipcode,
-              address1: element.location.address1,
-              address2: element.location.address2,
-            },
-            locationName: "",
+            // location: {
+            //   countryID: element.location.countryID,
```
Comment (kunalfleethawks) [inline src/app/pages/dispatch/trips/trip-detail/trip-detail.component.ts]: remove this code if not required.

**Review by kunalfleethawks — CHANGES_REQUESTED** [review]

**Review by FH-sandeepkaur1 — COMMENTED** [review]
Commented code is removed. Please review

**Review by kunalfleethawks — APPROVED** [review]

**Pull request merged**